[OTX] - 📝 Add documentation

samet-akcay opened the pull request:
This PR adds the latest documentation.

samet-akcay committed:
Merge develop and xfail dataset tests

samet-akcay committed:
Import label_chema_example in conftest.py

samet-akcay committed:
Added the docs

samet-akcay committed:
Added the docs

ashwinvaidya17 commented on requirements/docs.txt:
```diff
@@ -1,5 +1,7 @@
 sphinxemoji==0.1.8
-sphinx_rtd_theme
\ No newline at end of file
+sphinx_rtd_theme
```
Since we are using furo theme maybe we can remove read the docs

ashwinvaidya17 commented on otx/api/entities/__init__.py:
```diff
@@ -1,4 +1,105 @@
-"""Shape and graph entities."""
+"""Shape and graph entities.
+
+.. automodule:: otx.api.entities.annotation
```
Is it a common practice to add automodule to `__init__.py`. Can't we move it to `docs/apis/entities.rst` since we already have that?

samet-akcay replied:
I moved all of them to the rst files

ashwinvaidya17 commented on ote_sdk/ote_sdk/usecases/exportable_code/demo/LICENSE:
```diff
@@ -1,2 +1,201 @@
-Copyright (C) 2021-2022 Intel Corporation
-SPDX-License-Identifier: Apache-2.0
+Apache License
```
Is this needed? Isn't it part of ote_sdk?

samet-akcay replied:
I did pull the development branch. When we merge this to develop, this change wont be picked up

ashwinvaidya17 commented on docs/source/index.rst:
```diff
@@ -1,15 +1,19 @@
+.. training_extensions documentation master file, created by
+   sphinx-quickstart on Fri Sep  9 09:58:44 2022.
    You can adapt this file completely to your liking, but it should at least
    contain the root `toctree` directive.
```
Is this autogenerated each time?

samet-akcay replied:
nope, just left over I think

ashwinvaidya17 reviewed (commented):
Thanks for the efforts, I have a few comments. Mostly, I am concerned about having sphinx stuff in __init__ files

samet-akcay committed:
Move automodule from init to rst files

samet-akcay committed:
Remove autogenerated comment from index.rst file

samet-akcay committed:
Remove sphinx-rtd-theme from the requirements

samet-akcay commented:
> Thanks for the efforts, I have a few comments. Mostly, I am concerned about having sphinx stuff in init files

Thanks for the review. I think I've addressed all of your comments now.

ashwinvaidya17 reviewed (approved):
Thanks!

samet-akcay commented:
> Thanks!

Weirdly, the tests took 42 mins and failed :/
